server/server.go

```diff
@@ -682,8 +710,8 @@ func (s *Server) registerAccount(acc *Account) {
 	// already created (global account), so use locking and
 	// make sure we create only if needed.
 	acc.mu.Lock()
-	if acc.rm == nil && s.opts != nil && (s.opts.Cluster.Port != 0 || s.opts.Gateway.Port != 0) {
-		acc.rm = make(map[string]*rme, 256)
+	if acc.rm == nil && s.opts != nil && s.shouldTrackSubscriptions() {
```

Comment:
I am pretty sure we don't need that for Gateways.. my guess is that I did in an early iteration of implementing gateways think that I could use that, but ending up having to have my own storage and forgot to remove this condition.

Reply:
I will add a comment but leave code for now.